Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanFragment.kt

```diff
@@ -111,56 +101,45 @@ class SubmissionQRCodeScanFragment : Fragment(R.layout.fragment_submission_qr_co
-                    CoronaTestResult.PCR_NEGATIVE,
-                    CoronaTestResult.PCR_INVALID,
-                    CoronaTestResult.PCR_REDEEMED ->
-                        doNavigate(
-                            NavGraphDirections
-                                .actionSubmissionTestResultPendingFragment(
-                                    testType = CoronaTest.Type.PCR
-                                )
-                        )
-                    CoronaTestResult.RAT_POSITIVE ->
-                        doNavigate(
-                            NavGraphDirections
-                                .actionToSubmissionTestResultAvailableFragment(
-                                    testType = CoronaTest.Type.RAPID_ANTIGEN
-                                )
-                        )
-                    CoronaTestResult.RAT_NEGATIVE,
-                    CoronaTestResult.RAT_INVALID,
-                    CoronaTestResult.RAT_PENDING,
-                    CoronaTestResult.RAT_REDEEMED ->
-                        doNavigate(
-                            NavGraphDirections
-                                .actionSubmissionTestResultPendingFragment(
-                                    testType = CoronaTest.Type.RAPID_ANTIGEN
-                                )
-                        )
+                return@observe2
+            }
+
+            when (state.test?.testResult) {
+                CoronaTestResult.PCR_POSITIVE -> {
+                    NavGraphDirections
+                        .actionToSubmissionTestResultAvailableFragment(testType = CoronaTest.Type.PCR)
+                        .run { doNavigate(this) }
+                }
+
+                CoronaTestResult.PCR_OR_RAT_PENDING -> {
+                    NavGraphDirections
+                        .actionSubmissionTestResultPendingFragment(testType = state.test.type)
+                        .run { doNavigate(this) }
+                }
+
+                CoronaTestResult.PCR_NEGATIVE,
+                CoronaTestResult.PCR_INVALID,
+                CoronaTestResult.PCR_REDEEMED -> {
+                    NavGraphDirections
+                        .actionSubmissionTestResultPendingFragment(testType = CoronaTest.Type.PCR)
+                        .run { doNavigate(this) }
+                }
+
+                CoronaTestResult.RAT_POSITIVE -> {
+                    NavGraphDirections
+                        .actionToSubmissionTestResultAvailableFragment(testType = CoronaTest.Type.RAPID_ANTIGEN)
+                        .run { doNavigate(this) }
+                }
+
+                CoronaTestResult.RAT_NEGATIVE,
+                CoronaTestResult.RAT_INVALID,
+                CoronaTestResult.RAT_PENDING,
+                CoronaTestResult.RAT_REDEEMED -> {
+                    NavGraphDirections
+                        .actionSubmissionTestResultPendingFragment(testType = CoronaTest.Type.RAPID_ANTIGEN)
+                        .run { doNavigate(this) }
```

Comment:
can we simplify it to ?
```kotlin
            val testResult = state.test?.testResult ?: return@observe2
            when (testResult) {
                CoronaTestResult.PCR_POSITIVE ->
                    NavGraphDirections.actionToSubmissionTestResultAvailableFragment(testType = CoronaTest.Type.PCR)

                CoronaTestResult.PCR_OR_RAT_PENDING ->
                    NavGraphDirections.actionSubmissionTestResultPendingFragment(testType = state.test.type)

                CoronaTestResult.PCR_NEGATIVE,
                CoronaTestResult.PCR_INVALID,
                CoronaTestResult.PCR_REDEEMED ->
                    NavGraphDirections.actionSubmissionTestResultPendingFragment(testType = CoronaTest.Type.PCR)

                CoronaTestResult.RAT_POSITIVE ->
                    NavGraphDirections.actionToSubmissionTestResultAvailableFragment(testType = CoronaTest.Type.RAPID_ANTIGEN)

                CoronaTestResult.RAT_NEGATIVE,
                CoronaTestResult.RAT_INVALID,
                CoronaTestResult.RAT_PENDING,
                CoronaTestResult.RAT_REDEEMED ->
                    NavGraphDirections.actionSubmissionTestResultPendingFragment(testType = CoronaTest.Type.RAPID_ANTIGEN)
            }.run { doNavigate(this) }
```

Reply:
Yes, see  https://github.com/corona-warn-app/cwa-app-android/pull/3150/commits/2e53a340ed469c4829490f2208e55d388137eb07